engine/classes/Elgg/ViewsService.php

```diff
@@ -89,37 +89,73 @@ public function __construct(\Elgg\PluginHooksService $hooks, \Elgg\Logger $logge
 			'fallback' => [],
 		];
 	}
+	
+	/**
+	 * Takes a view name and returns the canonical name for that view.
+	 * 
+	 * @param string $alias The possibly non-canonical view name.
+	 * 
+	 * @return string The canonical view name.
+	 */
+	public function canonicalizeViewName(/*string*/ $alias) /*: string*/ {
+		if (!is_string($alias)) {
+			return false;
+		}
+		
+		$canonical = $alias;
+
+		if (strpos($canonical, "js/") === 0) {
+			$canonical = substr($canonical, 3);
+			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
```

Comment:
So, I think this will catch views like jquery.ui or jquery.form and think they have an extension like "ui" or "form", and therefore fail to add ".js". We should check the extensions against a whitelist to make sure they are recognized, then move them.